packages/components/form/src/form.ts

```diff
@@ -177,7 +177,10 @@ export class Form<T extends Record<string, any> = Record<string, any>> extends L
   }
 
   #onFormControl(event: SlFormControlEvent): void {
-    const control = event.target;
+    if (!(event.composedPath()[0] instanceof EventTarget) || !(event.composedPath()[0] instanceof HTMLElement)) {
```

Comment:
I would add type guard below that method, sth like:

```
#isFormControl(element: unknown): element is HTMLElement & FormControl {
    return element instanceof HTMLElement && 'formControlElement' in element;
  }
  ```
  
  and here it could look like 
  ```suggestion
    if (
      !(event.composedPath()[0] instanceof EventTarget) ||
      !(event.composedPath()[0] instanceof HTMLElement) ||
      !this.#isFormControl(event.composedPath()[0])
    ) {
```